kura/org.eclipse.kura.nm/src/main/java/org/eclipse/kura/nm/status/NMStatusConverter.java

```diff
@@ -60,81 +61,109 @@ public class NMStatusConverter {
         builder.withState(deviceStateConvert(deviceState));
 
         builder.withUsbNetDevice(usbNetDevice);
 
-        setDeviceStatus(builder, deviceProperties.getDeviceProperties());
+        setDeviceStatus(builder, devicePropertiesWrapper);
         setIP4Status(builder, ip4configProperties);
-        setWifiStatus(builder, deviceProperties.getWirelessDeviceProperties(),
+        setWifiStatus(builder, devicePropertiesWrapper.getDeviceSpecificProperties().get(),
                 accessPointsProperties.getActiveAccessPoint(), accessPointsProperties.getAvailableAccessPoints(),
                 supportedChannelsProperties.getCountryCode(), supportedChannelsProperties.getSupportedChannels());
 
         return builder.build();
     }
 
-    private static void setDeviceStatus(NetworkInterfaceStatusBuilder<?> builder, Properties deviceProperties) {
-        builder.withAutoConnect(deviceProperties.Get(NM_DEVICE_BUS_NAME, "Autoconnect"));
-        builder.withFirmwareVersion(deviceProperties.Get(NM_DEVICE_BUS_NAME, "FirmwareVersion"));
-        builder.withDriver(deviceProperties.Get(NM_DEVICE_BUS_NAME, "Driver"));
-        builder.withDriverVersion(deviceProperties.Get(NM_DEVICE_BUS_NAME, "DriverVersion"));
+    private static void setDeviceStatus(NetworkInterfaceStatusBuilder<?> builder,
+            DevicePropertiesWrapper devicePropertiesWrapper) {
+        builder.withAutoConnect(devicePropertiesWrapper.getDeviceProperties().Get(NM_DEVICE_BUS_NAME, "Autoconnect"));
+        builder.withFirmwareVersion(
+                devicePropertiesWrapper.getDeviceProperties().Get(NM_DEVICE_BUS_NAME, "FirmwareVersion"));
+        builder.withDriver(devicePropertiesWrapper.getDeviceProperties().Get(NM_DEVICE_BUS_NAME, "Driver"));
+        builder.withDriverVersion(
+                devicePropertiesWrapper.getDeviceProperties().Get(NM_DEVICE_BUS_NAME, "DriverVersion"));
 
-        UInt32 mtu = deviceProperties.Get(NM_DEVICE_BUS_NAME, "Mtu");
+        UInt32 mtu = devicePropertiesWrapper.getDeviceProperties().Get(NM_DEVICE_BUS_NAME, "Mtu");
         builder.withMtu(mtu.intValue());
 
-        String hwAddress = deviceProperties.Get(NM_DEVICE_BUS_NAME, "HwAddress");
-        builder.withHardwareAddress(getMacAddressBytes(hwAddress));
+        try {
+            String hwAddress = devicePropertiesWrapper.getDeviceProperties().Get(NM_DEVICE_BUS_NAME, NM_HW_ADDRESS);
+            builder.withHardwareAddress(getMacAddressBytes(hwAddress));
+        } catch (DBusExecutionException e) {
+            logger.debug("NetworkManager version equal or lower then 1.24 detected.");
+            switch (devicePropertiesWrapper.getDeviceType()) {
+            case NM_DEVICE_TYPE_ETHERNET:
+                builder.withHardwareAddress(getMacAddressBytes(
+                        devicePropertiesWrapper.getDeviceProperties().Get(NM_DEVICE_WIRED_BUS_NAME, NM_HW_ADDRESS)));
```

Comment:
Did you notice that you didn't call the correct method here (i.e. `getDeviceSpecificProperties`) and it works anyway? 😄 

We noticed this behaviour only recently: when you call a `Get` it behaves sort of like a cast to the objec type represented by the BUS name.

So... if you have the `Device` properties, you can perform a `Get` to the `Device.Wireless` properties, without carrying around the specific `Properties` themselves.

RTFM I guess... 😆 

For now let's keep using the `Wrapper` and the device specific properties. We'll think about refactoring the `NMStatusConverter`using this behaviour later (for example for when we'll have the unit tests for the converter... see #4418 )